Part 6 (#26)
Svelte-Society-Fr/kit: i18n.md

# i18n

You'll find in this section some information about the process we used to translate the Svelte docs to french. You can check this [draft PR](https://github.com/sveltejs/kit/pull/10817) to get an idea of the different files we modified.

You can find the translated version at the following address: https://kit.sveltefr.dev.

## Fork

**We recommend that you fork the [original SvelteKit repo](https://github.com/sveltejs/kit)** rather than this one to keep the original repo as reference.

> Either way, you should know that the main modifications we made to original repo are the following :

Reviewer: ```suggestion
Either way, you should know that the main modifications we made to the original repo are the following:
```

- `sites/kit.svelte.dev/src/routes/+page.svelte`
- `sites/kit.svelte.dev/src/routes/docs/[slug]/+page.svelte`
- `sites/kit.svelte.dev/src/routes/home/Intro.svelte`

> Be careful, this list might not be exhaustive.

### Markdown processing

As we have seen in the previous section, the docs contains references to external sites. You might want to also reference these links in the `.md` files, but this will require to **replace the environment variables at build time**.

This can done in the `sites/kit.svelte.dev/src/lib/server/docs/index.js` file, like so:

```js
// ...
const content = page.content
	.replace(/PUBLIC_SVELTE_SITE_URL/g, PUBLIC_SVELTE_SITE_URL)
	.replace(/PUBLIC_KIT_SITE_URL/g, PUBLIC_KIT_SITE_URL)
	.replace(/PUBLIC_LEARN_SITE_URL/g, PUBLIC_LEARN_SITE_URL)
	.replace(/PUBLIC_GITHUB_ORG/g, PUBLIC_GITHUB_ORG);
// ...
```

## Translating the docs

The core of the work to accomplish is in the `documentation/docs` folder.

**We translated as much as possible in this folder**, even some strings in the code snippets so that even code examples are better understood.

Do not forget to translate the `meta.json` files, which provide metadata like section titles.

> We did not feel necessary to translate the folder names, since it would 'only' affect URLs. This might have an bad influence on SEO. However, **the subsections titles in the `.md` files that you will eventually translate will break links** throughout your site. You will need to fix those before deployment.

Reviewer: ```suggestion
We did not feel necessary to translate the folder names, since it would 'only' affect URLs. This might have a bad influence on SEO. However, **the subsections titles in the `.md` files that you will eventually translate will break links** throughout your site. You will need to fix those before deployment.
```

Here's the list of files in which you might need to translate UI stuff:
- `sites/kit.svelte.dev/src/routes/+error.svelte`
- `sites/kit.svelte.dev/src/routes/+layout.server.js`
- `sites/kit.svelte.dev/src/routes/+layout.svelte`
- `sites/kit.svelte.dev/src/routes/+page.svelte`
- `sites/kit.svelte.dev/src/routes/docs/[slug]/+page.svelte`
- `sites/kit.svelte.dev/src/routes/home/Deployment.svelte`
- `sites/kit.svelte.dev/src/routes/home/Features.svelte`
- `sites/kit.svelte.dev/src/routes/home/Hero.svelte`
- `sites/kit.svelte.dev/src/routes/home/Intro.svelte`
- `sites/kit.svelte.dev/src/routes/home/Showcase.svelte`
- `sites/kit.svelte.dev/src/routes/home/Svelte.svelte`
- `sites/kit.svelte.dev/src/routes/home/subtitles.vtt`
- `sites/kit.svelte.dev/src/routes/nav.json/+server.js`
- `sites/kit.svelte.dev/src/routes/schema.json`
- `sites/kit.svelte.dev/src/routes/search/+page.svelte`

> Be careful, this list might not be exhaustive.

Additionally, you also might want to change the `lang` attribute in `sites/svelte.dev/src/app.html`.

## Modules and interfaces

> The 'Modules' and 'Types' sections of the docs directly refer to the kit package source code. For instance, the documentation [here](https://kit.svelte.dev/docs/modules#$app-environment-building) and [here](https://kit.svelte.dev/docs/types#public-types-adapter) is not written anywhere in the `documentation` folder, but in the kit package source code instead.

Reviewer: ```suggestion
The 'Modules' and 'Types' sections of the docs directly refer to the kit package source code. For instance, the documentation [here](https://kit.svelte.dev/docs/modules#$app-environment-building) and [here](https://kit.svelte.dev/docs/types#public-types-adapter) are not written anywhere in the `documentation` folder, but in the kit package source code instead.
```

> If you want the types description to also be translated, then you will need to work on these files:

Reviewer: ```suggestion
If you want the type descriptions to also be translated, then you will need to work on these files:
```

- `packages/kit/src/exports/public.d.ts`
- `packages/kit/src/exports/vite/index.js`
- `packages/kit/src/runtime/app/environment.js`
- `packages/kit/src/runtime/app/forms.js`
- `packages/kit/src/runtime/app/navigation.js`
- `packages/kit/src/runtime/app/stores.js`
- `packages/kit/src/types/ambient.d.ts`
- `packages/kit/src/types/private.d.ts`
- `packages/kit/src/types/synthetic/$env+dynamic+private.md`
- `packages/kit/src/types/synthetic/$env+dynamic+public.md`
- `packages/kit/src/types/synthetic/$env+static+private.md`
- `packages/kit/src/types/synthetic/$env+static+public.md`
- `packages/kit/src/types/synthetic/$lib.md`
- `sites/kit.svelte.dev/scripts/types/index.js`

> Be careful, this list might not be exhaustive.

> In these `packages` files, you probably won't be able to use the environment variables for links described [here](#external-links), since it's a different package. It might actually be possible, but we did not investigate too much on the matter.

> You will need to run `npm run generate:types` in the `packages/kit` folder before running `npm run dev` in the `sites/kit.svelte.dev` folder in order to be able to see your changes in dev mode.

## Glossary

> In a web development context, some words do not translate well in some languages – or at least in french –, even if they technically have an official translation. This is because people have taken the habit to use the original word in english rather than the translated one. These are called _anglicisms_.

Reviewer: ```suggestion
In a web development context, some words do not translate well in some languages – or at least in French –, even if they technically have an official translation. This is because people have taken the habit to use the original word in English rather than the translated one. These are called _anglicisms_.
```

For example, in french, _callback_ should be translated to _fonction de rappel_. But in practice almost nobody uses the translated word, and we stick to _callback_ instead.

For this reason, **we chose to remain as close as possible to the actual usage**, and keep such words in their original version.

> But such anglicisms might not be fully understood by some people, especially beginners in tech. So we added an [extra glossary section](documentation/docs/07-glossary) in the docs in order to provide a simple definition of the anglicisms.

Reviewer: c'est la bonne url ? Ce n'est pas sur le site de svelte ?

Author: Ce n'est pas pertinent, ct une relique de mon rebase raté

**Please note that the anglicisms used in this repo are linked to Svelte docs site**, using the `PUBLIC_SVELTE_SITE_URL` variable.

The glossary is divided in four sections:
- `01-development`: generic development-related words
- `02-web`: web-related words
- `03-javascript`: javascript-related words
- `04-sveltejs`: svelte-related words

We tried not to be too specific in the definitions of this glossary, providing instead links to Wikipedia or MDN.

> > Depending on your language and on how people use anglicisms for technical words, your glossary sections might be very different from the french one, or **you might not even need such glossary**. It's up for you to decide really.

Reviewer: ```suggestion
> Depending on your language and on how people use anglicisms for technical words, your glossary sections might be very different from the French one, or **you might not even need such glossary**. It's up to you to decide really.
```

> We are aware that the SvelteKit site has its own Glossary section, in which it defines concepts like SSR, which we also defined in the Svelte docs glossary. In the future we might move the definitions of these concepts that we defined in the Svelte docs to this SvelteKit docs site.

### Styling links

To differentiate the anglicisms links from regular links, we added some styling in `sites/kit.svelte.dev/src/routes/+layout.svelte`:

```css
:global(.text .vo a) {
	color: var(--sk-text-1);
	box-shadow: inset 0 -1px 0 0 var(--sk-text-4);
	transition: color 0.2s ease-in-out;
}

:global(.text .vo a:hover) {
	color: var(--sk-text-3);
}
```

> > `'vo'` stands for "original version" (_version originale_) in french, and is what we use to describe movies that aren't dubbed.

Reviewer: ```suggestion
> `'vo'` stands for "original version" (_version originale_) in french, and is what we used to describe movies that aren't dubbed.
```
